lib/linux_admin/process/memory.rb

```diff
@@ -0,0 +1,49 @@
+module LinuxAdmin
+  module Process
+    class Memory
+      attr_reader :pss, :rss, :uss, :shared, :swap, :size
+
+      def initialize(pid = Process.pid, smaps = nil)
```

Comment:
I'm not sure I like the dependency injection as part of the `.new`, because it binds you to that signature for the future and would require a major bump to change it.  Say, tomorrow a different distribution uses a different file, then you have to add a new param, and the signature is weird because callers have to pass a nil, etc.  Perhaps allow smaps_data to be set, and if set, then return it otherwise look it up...something like

```ruby
def initialize(pid = Process.pid)
  @pid = pid
  ...
end

attr_writer :smaps_data

def smaps_data
  @smaps_data ||= begin
    smaps = "/proc/#{pid}/smaps"
    raise "smaps not found: #{smaps}" unless File.exist?(smaps)
    File.read(smaps)
  end
end

# Then in the specs:

subject do
  described_class.new.tap { |o| o.smaps_data = @smaps }
end
```

I realize that changes the timing of the parse call, but maybe that shouldn't be on the `.new`?

Another option is to allow initialize to take a block and allow configuration that way?

```ruby
def initialize(pid = Process.pid)
  @pid = pid
  @pss = 0
  ...
  yield self if block_given?
  parse_smaps
end

# In spec

subject do
  described_class.new { |o| o.smaps_data = @smaps }
end